### PR TITLE
fix(gradle): Fix gradle example and docs to work with java 12+

### LINKED ISSUE #702: Gradle: Set java target level in cookiecutter and docs
**What were you trying to accomplish?**

Building the gradle project using jdk 17

## Expected Behavior

Project should compile with target level of java 11 or java 1.8

## Current Behavior

Sam template example fails

```
Running JavaGradleWorkflow:GradleBuild

Build Failed
Error: JavaGradleWorkflow:GradleBuild - Gradle Failed: [error] target level should be in '1.1'...'1.8','9'...'16' (or '5.0'..'16.0') or cldc1.1: 17
```

## Possible Solution

Update gradle example

```gradle
plugins{
    id 'java'
    id 'io.freefair.aspectj.post-compile-weaving' version '6.3.0'
}

repositories {
    mavenCentral()
}

dependencies {
    aspect 'software.amazon.lambda:powertools-logging:1.10.2'
    aspect 'software.amazon.lambda:powertools-tracing:1.10.2'
    aspect 'software.amazon.lambda:powertools-metrics:1.10.2'

    implementation 'com.amazonaws:aws-lambda-java-core:1.2.1'
    implementation 'com.amazonaws:aws-lambda-java-events:3.11.0'

    implementation 'org.apache.logging.log4j:log4j-api:2.17.1'
    implementation 'org.apache.logging.log4j:log4j-core:2.17.1'

    testImplementation 'junit:junit:4.13.2'
}

sourceCompatibility = 11
targetCompatibility = 11

```

## Steps to Reproduce (for bugs)
1. `sam init --location gh:aws-samples/cookiecutter-aws-sam-powertools-java`
2. Select gradle and java11 (with java 17 installed locally)
3. run `sam build`

## Environment

* **Powertools version used**: 1.10.2
* **Packaging format (Layers, Maven/Gradle)**: Gradle
* **AWS Lambda function runtime:** Java 11 (or Java 1.8)

### PR DESCRIPTION
**Pull request by michaelbrewer**
**Issue #, if available:**

closes #702

## Description of changes:

Changes:
- Bump gradle version to 7.3.3
- Use `io.freefair.aspectj.post-compile-weaving` in example project
- Specify java 11 for sourceCompatibility, targetCompatibility
- Bump junit version to 4.13.2

closes #702

**Checklist**

 

* [X] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-java/#tenets)
* [X] Update tests
* [X] Update docs
* [X] PR title follows [conventional commit semantics]()

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by pankajagrawal16 — DISMISSED** [review]

File: docs/index.md [hunk 1]
```diff
@@ -117,7 +117,13 @@ For more information about the project and available options refer to this [repo
         aspect 'software.amazon.lambda:powertools-logging:{{ powertools.version }}'
         aspect 'software.amazon.lambda:powertools-tracing:{{ powertools.version }}'
         aspect 'software.amazon.lambda:powertools-metrics:{{ powertools.version }}'
+        implementation 'software.amazon.lambda:powertools-logging:{{ powertools.version }}'
```
Comment (pankajagrawal16) [inline docs/index.md:120]: is implementation needed ?

Comment (michaelbrewer) [reply docs/index.md:120]: maybe not, i saw it was in the cookiecutter examples. it works without it.

Comment (pankajagrawal16) [reply docs/index.md:120]: Hmm i thought i removed it from all the usage in cookiecutter. I will check and fix that if its still present.

Comment (pankajagrawal16) [reply docs/index.md:120]: I checked, its not present in cookiecutter project as well.

Comment (michaelbrewer) [reply docs/index.md:120]: cool. i will remove it then :)

**Review by pankajagrawal16 — APPROVED** [review]

**Pull request merged**